### ISSUE TITLE
Regressing Cell Cycle genes in ScaleData step or while running Harmony using vars_use? 

### ISSUE THREAD

**Issue by goncik** [immunogenomics/harmony#117]
Hi, 

I have 3 different 10X single-cell RNA-seq dataset that I want to integrate using Harmony. They are all in same condition, but just coming from different labs. When I ran Harmony, I saw some clusters separated with cell cycle genes. Therefore I want to regress out the cell cycle effect from my data. 

However, I am a bit confused on which step to do that; either on ScaleData() step before running harmony or while running RunHarmony() as a covariate to integrate. Which one would make sense? 

I'm not sure but ScaleData() step is still not batch-corrected yet and I'm afraid to cause more trouble by doing regression on that step. 

Many thanks! 
G

**Comment by michael-vermeulen** [comment]
I use Harmony frequently to batch correct single cell data. 
I combine datasets and regress unwanted variables (eg. nUMI, mito.percent) using ScaleData before Harmony correction.  I would add cell cycle correction in an additional ScaleData step before Harmony. 
You can can then view how the batches/datasets are mixed before and after Harmony correction to get a general idea of 
how it went. You can then try adding cell cycle as a Harmony var and see how that goes. I would guess that the latter would alter the structure of your data more than you'd like but there's no harm in testing it out.

**Comment by ilyakorsunsky** [comment]
Dear @goncik and @michael-vermeulen, 

Unfortunately, Harmony does not currently support integration over continuous variables. For now, if you want to "hack" harmony to regress out percent mitochondrial reads, e.g., I would bin your continuous variable into a categorical one and Harmonize over the binned categorical variable. The code would look something like this: 

```
obj@meta.data$percent.mito.bin <- cut(obj@meta.data$percent.mito, breaks=10)
obj <- runHarmony(obj, "percent.mito.bin")
```

Let me know how it goes! 

Best, 
Ilya

**Comment by vertesy** [comment]
1. Is `Harmony does not currently support integration over continuous variables.` still the case?
2. Does that work in Seurat v5's implementation of Harmony, or it can't be done no matter the implementation?

**Comment by pati-ni** [comment]
Hi @vertesy

1.  yes this is still the case. You can follow @ilyakorsunsky suggestion as a workaround.

2. I would advice against Seurat's wrapper. Last time I checked it does only one level correction.

**Comment by vertesy** [comment]
Thank you, I tried that but without success: binning created too many small categories. 

On an object of around 25,000 HQ cells, I wanted to regress out 
- the 10 different samples, 
- a binary nucleus versus whole cell status, and
- cell cycle, as recommended in Seurat 

That is done by two scores S and G2, I have binned these, by then, then by 5, (maybe I went even lower).

And finally collapse all four of the above, into a single metadata col. 

Unfortunately in each case very small categories are created by one or a few cells, 

And Harmony failed during integration (regression).
